### PR TITLE
Pydantic Body/Requests

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
This PR adds REST models to the web web handler for non-queue interfaces. Provides both client and server side level of validation. Most of it is bare bones right now, but can be improved upon later.

## Status
- [x] Ready to go

edited by lnaden

### CODE REVIEW

**Comment by Lnaden** [comment]
Sorry about the sloppy git history there, i rebased from master to get the current master's changes in to get tests to trigger, we can clean up again after the fact

**Comment by Lnaden** [comment]
This is ready for formal review now, I think all the changes are in

File: qcfractal/interface/models/rest_models.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    class Meta(BaseModel):
+        projection: Dict[str, Any] = None
+
+    meta: Meta = Meta()
+    data: Dict[str, Any]
+
+    @validator("data", whole=True)
+    def only_data_keys(cls, v):
+        keys = ["program", "molecule", "driver", "method", "basis", "options", "hash_index", "task_id", "id", "status"]
+        final_data = {key: v[key] for key in keys if key in v}
```
Comment (dgasmith) [inline qcfractal/interface/models/rest_models.py]: Can we do something like:

```
valid_keys = {"program", ...}
final_data = {key: v[key] for key in (v.keys() & valid_keys)}
```

I feel like we miss out on set manipulation too much!

**Comment by dgasmith** [comment]
I will merge this, and handle the one issue later.

**Pull request merged**